### DISCUSSION TITLE
The difference between Windows versions in GitHub

### DISCUSSION THREAD

**Discussion by Enlight432** [xgi/houdoku#340] [Q&A] [answered]
Sorry, but I don't understand the difference between the three versions below
Houdoku-Setup-2.14.0.exe
Houdoku-2.14.0-win.zip
Houdoku-2.14.0.exe

**Comment by Enlight432** [comment]
I even asked some friends, they said it is not clear
Which one should I install ?

**Comment by Lockszmith-GH** [comment, marked as answer]
I'm not the maintainer, but I just took a look at the three files, and tested them out.

`Houdoku-Setup-#.#.#.exe` - this is an Installer (specifically a [Nullsoft installer](https://nsis.sourceforge.io/Main_Page))
`Houdoku-#.#.#-win.zip` - This is a portable folder.
`Houdoku-#.#.#.exe` - This is a portable package, it extracts itself into a temporary folder, and runs from there.

If you don't like to mess around with the technicalities of application management, and are ok with installing applications on your system (which is most people), just run the **Setup**.
You might prefer to not deal with the download yourself and just run:
```powershell
winget install xgi.houdoku
```

If you usually like portable applications and feel that installers mess up/clutter your system, you probably want one of the portable options.
The **portable folder**, you just unzip into a directory, and run it from there.
The **portable package** will self extract itself into a temporary directory, and when you exit the app, that dir is destroyed.

The main difference between the portable folder and the package is time. The package extracts everytime you run, that means it takes time to do that. Unzipping once, you don't need to wait for the extraction.

Hope this helps.

**Reply by Enlight432** [reply]
@Lockszmith-GH 

Thank you very much for the answer 
I'm curious Why when people tend to use "Setup" method, the "portable package" method is included in the "https://houdoku.org/download" site as the default installation method !?
Is there a particular problem with the "Setup" file? I don't think so, but it is strange

**Reply by Lockszmith-GH** [reply]
~Probably has to do with the fact that this app "isn't 100% above board" - if you know what I mean, it pulls from sites that are considered NSFW, whether for the content itself or the legitimacy of the content.
And so, running a portble version reduces risk.~

~Again, just a hunch, as I don't really know.~

UPDATE: Wrong hunch, read @xgi's answer below.

**Reply by xgi** [reply]
The houdoku.org page defaulting to the portable installer was not my intention, there's just code that picks up the first `.exe` file which made sense before I added the portable option. I'd generally recommend the installer (with Setup in the name) unless you have a good reason not to. I'll update the website so it defaults to that version.

And thanks for answering!

**Reply by Lockszmith-GH** [reply]
Thanks for looking into this @xgi, and thanks for making such an awesome application.
